lib/screens/events/events.dart

```diff
@@ -12,29 +13,30 @@ class EventsScreen extends StatefulWidget {
 class _EventsScreenState extends State<EventsScreen> {
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      body: BlocConsumer<EventsCubit, EventsState>(
-        listener: (context, state) {
-          if (state is EventsInitial) {
-            _getAllEvents();
-          } else if (state is EventsError) {
-            Scaffold.of(context).showSnackBar(
-              SnackBar(content: Text(state.message)),
-            );
-          }
-        },
-        builder: (context, state) {
-          if (state is EventsInitial) {
-            return _buildInitial();
-          } else if (state is EventsSuccess) {
-            return _buildSuccess(context, state.json);
-          } else if (state is EventsLoading) {
-            return _buildLoading();
-          } else {
-            Log.e(tag: "EventsState", message: "State now is EventsError reached");
-            return Container(); // TODO the user should be shown the error on screen instead of a snackbar, and a retry button.
-          }
-        },
+    return StatefulWrapper(
+      onInit: () => _getAllEvents(),
+      child: Scaffold(
+        body: BlocConsumer<EventsCubit, EventsState>(
+          listener: (context, state) {
+            if (state is EventsError) {
+              Scaffold.of(context).showSnackBar(
+                SnackBar(content: Text(state.message)),
+              );
+            }
+          },
+          builder: (context, state) {
+            if (state is EventsInitial) {
+              return _buildInitial();
+            } else if (state is EventsSuccess) {
+              return _buildSuccess(context, state.json);
+            } else if (state is EventsLoading) {
+              return _buildLoading();
+            } else {
+              Log.e(tag: "EventsState", message: "State now is EventsError reached");
+              return Container(); // TODO the user should be shown the error on screen instead of a snackbar, and a retry button.
```

Comment:
update on the retry screen?

Reply:
Had a talk with him (Mohit) yesterday he said he is working on it.